### ISSUE TITLE
'guru implements' returns 'guru failed' in the implements tab

### ISSUE THREAD

**Issue by veger** [joefitzgerald/go-plus#630]
### Description

The _guru_ command is installed by go-plus, I do not have it installed manually (as I do/did not use it)

When running _guru implements_, the developer console shows 
```
props running guru -json implements /home/user/go/src/etc/etc/my_file.go:#42

guru failed
```

Note that _golang:find-usages_ does work running
```
guru -json referrers /home/user/go/src/etc/etc/my_file.go:#42
```
as it properly shows the usages of the function I tried it on

### Output from `atom -v && apm -v`

Atom    : 1.16.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
apm  1.16.1
npm  3.10.5
node 6.9.5 x64
python 2.7.12+
git 2.9.3

### Output From `go env`

GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/user/devel/go"
GORACE=""
GOROOT="/usr/lib/go-1.7"
GOTOOLDIR="/usr/lib/go-1.7/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build936633827=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"

### Steps to Reproduce

1. Run _Golang:implements_ command in a go file. In my case I tried it on a `type MyInterface interface` definition.

**Edit:** Hm... when running it on _MyInterface_ in the following line it actually does work:
```
func NewMyInterface(i ImplementationType) (MyInterface, error) {
```

Is this expected? (If so, the returned result _'guru failed'_ is confusing in this case...) Or is the _guru implements_ command partially broken?

**Comment by zmb3** [comment]
The `-format` flag was changed to `-json` [a little over a year ago](https://github.com/golang/tools/commit/2da0720e4f0a8d82e510747c24a9e09038acb984).

It sounds like you just need to update, which you can do by either:
 - running `golang:update tools` in Atom
 - running `go get -u golang.org/x/tools/cmd/guru` in your shell

**Comment by veger** [comment]
I just installed Atom and go-plus a couple of days ago. So I would expect that go-plus had installed the newest version of guru...

Which it did, but the golang-golang-x-tools Ubuntu package had priority (with _/usr/bin/guru_). After removing the Ubuntu (16.10) package, the one from my go workspace is used and in the terminal (when copying the command from the developer console) it now works.

But I still get the _guru failed_ result when running the tool from Atom/go-plus. (which is weird as copying the exact command to the terminal does work?!).

The GOPATH setting is empty, so it should use my environment variable. Is there anything else that might influence the execution of the guru command?

**Comment by zmb3** [comment]
So you've removed the outdated version of `guru` and only have the new version remaining, but you're still getting the same error?  

go-plus will search `$GOPATH/bin` first, which should find the newer version.  You shouldn't need to do anything else.

This may require some debugging.  Any chance you're open to a screen share?

**Comment by veger** [comment]
I found the 'issue': After adding some logging (only saw later that I could have enabled it elsewhere) I got exitcode 124. After some more fiddling I though of timeouts, and indeed after increasing the timeout (to 15 seconds) in `runGuru()` (in _implements.js_) it all worked. (Later I saw in the executor code that 124 indeed is a time-out).

So, this issue is mainly due to my old hardware I guess... ;) But a more clear error, besides _guru failed_ would have been convenient. I'll create a PR for this.

Whether 10s timeout is too short or not, a timeouts are actually needed, and whether it needs to be configurable, is not for me to decide... ;)
